samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+            barrier = D3D12_RESOURCE_BARRIER.InitTransition(_renderTargets[unchecked((int)_frameIndex)], D3D12_RESOURCE_STATE_RENDER_TARGET, D3D12_RESOURCE_STATE_PRESENT);
+            _commandList->ResourceBarrier(1, &barrier);
+
+            ThrowIfFailed(nameof(ID3D12GraphicsCommandList.Close), _commandList->Close());
+        }
+
+        private void WaitForPreviousFrame()
+        {
+            // WAITING FOR THE FRAME TO COMPLETE BEFORE CONTINUING IS NOT BEST PRACTICE.
+            // This is code implemented as such for simplicity. The D3D12HelloFrameBuffering
+            // sample illustrates how to use fences for efficient resource usage and to
+            // maximize GPU utilization.
+
+            // Signal and increment the fence value.
+            var fence = _fenceValue;
+            ThrowIfFailed(nameof(ID3D12CommandQueue.Signal), _commandQueue->Signal(_fence, fence));
+            _fenceValue++;
+
+            // Wait until the previous frame is finished.
+            if (_fence->GetCompletedValue() < fence)
+            {
+                ThrowIfFailed(nameof(ID3D12Fence.SetEventOnCompletion), _fence->SetEventOnCompletion(fence, _fenceEvent));
+                _ = WaitForSingleObject(_fenceEvent, INFINITE);
+            }
+
+            _frameIndex = _swapChain->GetCurrentBackBufferIndex();
+        }
+
+        protected override void Dispose(bool isDisposing)
+        {
+            var swapChain = _swapChain;
+
+            if (swapChain != null)
+            {
+                _swapChain = null;
+                _ = swapChain->Release();
+            }
+
+            var device = _device;
+
+            if (device != null)
+            {
+                _device = null;
+                _ = device->Release();
+            }
+
+            for (var index = 0; index < FrameCount; index++)
+            {
+                var renderTarget = _renderTargets[index];
+
+                if (renderTarget != null)
+                {
+                    _renderTargets[index] = null;
+                    _ = renderTarget->Release();
+                }
+            }
+
+            var commandAllocator = _commandAllocator;
+
+            if (commandAllocator != null)
+            {
+                _commandAllocator = null;
+                _ = commandAllocator->Release();
+            }
+
+            var commandQueue = _commandQueue;
+
+            if (commandQueue != null)
+            {
+                _commandQueue = null;
+                _ = commandQueue->Release();
+            }
+
+            var rootSignature = _rootSignature;
+
+            if (rootSignature != null)
+            {
+                _rootSignature = null;
+                _ = rootSignature->Release();
+            }
+
+            var rtvHeap = _rtvHeap;
+
+            if (rtvHeap != null)
+            {
+                _rtvHeap = null;
+                _ = rtvHeap->Release();
+            }
```

Comment:
`_cbvHeap` needs to be disposed similarly to this

Reply:
done.